lib/basedriver/mjpeg.js

```diff
@@ -0,0 +1,84 @@
+import request from 'request';
+import log from './logger';
+import MJpegConsumer from 'mjpeg-consumer';
+import { Writable } from 'stream';
+
+/** Class which stores the last bit of data streamed into it */
+export default class MJpegStream extends Writable {
+
+  /**
+   * Create an MJpegStream
+   * @param {string} mJpegUrl - URL of MJPEG-over-HTTP stream
+   * @param {function} errorHandler - additional function that will be
+   * called in the case of any errors.
+   * @param {object} [options={}] - Options to pass to the Writable constructor
+   */
+  constructor (mJpegUrl, errorHandler, options = {}) {
+    super(options);
+    this.errorHandler = errorHandler;
+    this.url = mJpegUrl;
+    this.clear();
+  }
+
+  /**
+   * Get the base64-encoded version of the JPEG
+   * @returns {string}
+   */
+  get lastChunkBase64 () {
+    if (!this.lastChunk) {
+      return null;
+    }
+
+    return this.lastChunk.toString('base64');
+  }
+
+
+  /**
+   * Reset internal state
+   */
+  clear () {
+    this.request = null;
+    this.consumer = null;
+    this.lastChunk = null;
+    this.updateCount = 0;
+  }
+
+  /**
+   * Start reading the MJpeg stream and storing the last image
+   */
+  start () {
+    this.consumer = new MJpegConsumer();
+
+    const onErr = (err) => {
+      log.error(`Error getting MJpeg screenshot chunk: ${err}`);
+      this.errorHandler(err);
+    };
+
+    this.request = request(this.url);
+
+    this.request
+      .on('error', onErr)   // ensure we do something with errors
+      .pipe(this.consumer)  // allow chunking and transforming of jpeg data
+      .pipe(this);          // send the actual jpegs to ourself
+  }
+
+  /**
+   * Stop reading the MJpeg stream. Ensure we disconnect all the pipes and stop
+   * the HTTP request itself. Then reset the state.
```

Comment:
It might be useful to mark overridden method with `@Override`

Reply:
this method is not overridden